src/js/elements/menu.js

```diff
@@ -1,174 +1,176 @@
-import {SubIndicator} from '../dataobjects'
-import {checkIfSubCategoryHasChildren, checkIfCategoryHasChildren, Component} from '../utils'
+import Controller from '../controller';
+import { checkIfCategoryHasChildren, checkIfSubCategoryHasChildren, Component } from '../utils';
```

Comment:
Might want to consider adding these check functions into a model - not 100% certain it makes sense, but definitely not in utils

Reply:
Makes sense to move them to the model. Done.